### ISSUE TITLE
Add support for injecting static routes

### ISSUE THREAD

**Issue by williamdes** [osrg/gobgp#2860]
As you can see on the project https://github.com/thepacketgeek/bgpd-rs there is a very simple section to inject routes into sessions.

Is there a limitation on this project to implement injecting routes into bgp peers?

To avoid a dependency on frr to get the routes to announce.

**Comment by fujita** [comment]
You can add any routes to gobgpd via the gRPC APIs.

**Comment by EmilyShepherd** [comment]
> You can add any routes to gobgpd via the gRPC APIs.

Is there any documentation for this?

**Comment by floatingstatic** [comment]
@EmilyShepherd See https://github.com/osrg/gobgp/blob/master/docs/sources/lib.md for an example on adding a route using grpc. 

This is also something that can also be done with the `gobgp` cli utility. Ex:

```
gobgp global rib add 203.0.113.0/24 nexthop 192.0.2.1 origin igp aspath 200,100 community 100:200
gobgp global rib -a ipv6 add 2001:db8:8000::/34 nexthop 2001:db8::51 origin igp aspath 300,200,100 community 100:300
```

**Comment by smorris76** [comment]
Is there a reason why support for static routes in the config file does not exist? Injecting routes via API or CLI is fine but then requires something to monitor the state of the gobgp process/rib to ensure the routes get readed if the process restarts.

**Comment by fujita** [comment]
> Is there a reason why support for static routes in the config file does not exist? Injecting routes via API or CLI is fine but then requires something to monitor the state of the gobgp process/rib to ensure the routes get readed if the process restarts.

Because I don’t think it’s a good idea to support various families of NLRIs as a file format.
More broadly, I don’t think configuring software through files is a good idea in the first place.

**Comment by EmilyShepherd** [comment]
>  More broadly, I don’t think configuring software through files is a good idea in the first place.

Ok, but gobgp _does_ already support configuration via files, so while holding an opinion that it isn't a good idea is valid in general, isn't that a decision that's long since already been made for this project?

> Because I don’t think it’s a good idea to support various families of NLRIs as a file format.

Again, gobgp's config files basically already do. Policy configuration (ie what routes should / should not be accepted) is configurable via the config file... why not just add static routes too?

**Comment by fujita** [comment]
> > More broadly, I don’t think configuring software through files is a good idea in the first place.
> 
> Ok, but gobgp _does_ already support configuration via files, so while holding an opinion that it isn't a good idea is valid in general, isn't that a decision that's long since already been made for this project?

Yes, this was decided on the first day of the project: the future BGP implementation should be controlled through an RPC API instead of using file-based configuration as in the existing BGP implementations. It will be then be controlled over gRPC by orchestrator systems, such as those used in data center, WAN, etc.

This was one of the reasons we decided to start the new BGP implementation project.
 
> > Because I don’t think it’s a good idea to support various families of NLRIs as a file format.
> 
> Again, gobgp's config files basically already do. Policy configuration (ie what routes should / should not be accepted) is configurable via the config file... why not just add static routes too?

We simply used OpenConfig's format. OpenConfig BGP configuration supports static routes?
